parquet-common/src/main/java/org/apache/parquet/compression/FallbackCodecFactory.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.apache.parquet.compression;
+
+import org.apache.parquet.Preconditions;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+
+/**
+ * {@link CompressionCodecFactory} implementation that delegates to the {@link CompressionCodecFactory} implementations it was constructed with.
+ */
+public class FallbackCodecFactory implements CompressionCodecFactory {
```

Comment:
It seems the usage for this is people don't care about which compression codec is used.

Reply:
Not necessarily don't care, but who might not have guarantees that any system might be able to run their preferred codecs. If you are to implement one using hardware acceleration there are platforms for which this might not work perfectly. This factory provides an easy and modular way to allow for more stable fallbacks for these scenarios OOTB.

Reply:
I agree with @ConeyLiu. Users can anyway do the same thing all by themselves. It seems sufficient when parquet reader and writer support passing a customized codec factory.